clients/src/main/java/org/apache/kafka/clients/Metadata.java

```diff
@@ -161,6 +169,51 @@ public synchronized int requestUpdate() {
         return this.version;
     }
 
+    /**
+     * Request an update for the partition metadata iff the given leader epoch is at newer than the last seen leader epoch
+     */
+    public synchronized boolean updateLastSeenEpochIfNewer(TopicPartition topicPartition, int leaderEpoch) {
+        Objects.requireNonNull(topicPartition, "TopicPartition cannot be null");
+        if (updateLastSeenEpoch(topicPartition, leaderEpoch, oldEpoch -> leaderEpoch > oldEpoch) == leaderEpoch) {
+            this.needUpdate = true;
```

Comment:
Does it make sense to move this into `updateLastSeenEpoch`? It seems like we'd want to request an update in the `OffsetAndMetadata` overload as well. I guess the update logic also uses `updateLastSeenEpoch` though since maybe we need a flag or something.

Reply:
That does make sense. The logic of this method evolved a bit during this PR. At this point there's no reason we need to return the updated epoch for further logic -- we can just do the update and return a boolean and optionally update the flag.